docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,166 @@
+This encapsulates slightly more advanced filter logic over the existing clusterDeployment labels. In the example above, `hive_foo_counter` and `hive_foo_gauge` metrics will only be reported for the clusterDeployments that are labelled with aro-snowflake and not in limited support.
+
+### Implementation Details / Notes
+
+- All the options configured for a `metricsToReport` entry will work in tandem with each other for each metric listed in its `metricNames`. For ex, if all possible options are specified for a metric, then that metric will be reported with the additional labels as per additionalClusterDeploymentLabels, and will be reported only if it matches the labels and/or expressions as per clusterDeploymentLabelSelector and if the duration to be reported exceeds the minimumDuration.
+- This design allows for grouping of metrics for which you would want to define particular customizations. Grouping allows hiveConfig to be parsable. To avoid ambiguity, we will not allow multiple entries for any hive metric within `metricsToReport`.
+- We would be removing the fixed labels that can be propagated from the clusterDeployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.metricsToReport[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new metricsToReport, sufficient documentation should help though.
+- We would be renaming `hive_cluster_deployment_install_failure_total` and `hive_cluster_deployment_install_success_total` to ensure it ends with `seconds`, as it is a duration-based metric. Note that this will result in cloning the existing metric for the deprecation period, because if `metricsToReport` isn't present in the hiveConfig, then the _old_ way has to still work.
+- We would no longer panic when the metricsConfig fails validation, instead we would be updating hiveConfig.Status.Conditions[ReadyCondition] to false, with appropriate reason and message. See [Failure Modes](#failure-modes) for more details.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveConfig.spec.metricsConfig.metricsWithDuration and hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveConfig, you can either provide the new configurations (aka `metricsToReport`) or the old way of hiveConfig.spec.metricsConfig.metricsWithDuration and hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels. If there's an empty metricsConfig, we would default to _old_ behaviour for the period of deprecation.
```

Comment:
- I suppose conversion is not the write word, I'm just stating you can either provide the new `metricsToReport` option in the config, or follow whatever is implemented right now, to avoid ambiguity.
- Correct, but I also wanted to point out that all the metrics will become optional with `metricsToReport`. So when you change your HiveConfig to use the new options, you would also have to compile a list of all the hive metrics you use or need, and you'll have to configure them in the hiveConfig, otherwise hive will not report them.